docs/dev_guide/logger.rst

```diff
@@ -0,0 +1,77 @@
+.. _dev_logger:
+
+*********************************
+Logging, Warnings, and Exceptions
+*********************************
+
+Overview
+--------
+
+Sunpy makes use of a logging system to deal with messages (see :ref:`logger`). This provides the users and
+developers the ability to decide which messages to show, to capture them, and to optionally also send
+them to a file. The logger will log all messages issued directly to it but also warnings issued
+through warnings.warn() as well as exceptions.
+
+The logger is configured as soon as sunpy is imported. You can access it
+by importing it explicitly::
+
+    from sunpy import log
+
+Messages can be issued directly to it with the following levels and in the following way::
+
+    log.debug("Detailed information, typically of interest only when diagnosing problems.")
+
+    log.info("A message conveying information about the current task, and confirming that
+              things are working as expected.")
+
+    log.warning("An indication that something unexpected happened, or indicative of
+                 some problem in the near future (e.g. ‘disk space low’).
+
+                 The software is still working as expected.")
+    log.error("Due to a more serious problem, the software has not been able to
+               perform some function but the task is still continuing.")
+
+    log.critical("A serious error, indicating that the program itself may be unable to
+                  continue running. A real error may soon by issued and the task will fail.")
+
+The difference between logging a warning/error/critical compared to issuing a Python warning or raising
+an exception are subtle but important.
+
+Use Python `warnings.warn()` in library code if the issue is avoidable and the user code should be
+modified to eliminate the warning.
+
+Use `log.warning()` if there is likely nothing the user can do about the situation, but the event
+should still be noted. An example of this might be if the input data are all zeros. This may be unavoidable or
+even by design but you may want to let the user know.
+
+True exceptions (not `log.error()`) should be raised only when there is no way for the function to proceed.
+
+Regardless of the type (log message or warning or exception) messages should be one or two complete sentences
+that fully describe the issue and end with a period.
+
+Issuing Warnings
+----------------
+Sunpy warnings are provided by the :mod:`sunpy.util` module. The primary warning which
+should be used is :class:`sunpy.util.exceptions.SunpyUserWarning`. For deprecation use :class:`sunpy.util.exceptions.SunpyDeprecationWarning` or
+:class:`sunpy.util.exceptions.SunpyPendingDeprecationWarning`.
```

Comment:
9/10 what people should actually be using for deprecation warnings is `sunpy.util.decorators.deprecated`. I would suggest you re-write this section to put that option first and discuss the classes of warnings after.